modules/services/sleepwatcher/default.nix

```diff
@@ -0,0 +1,155 @@
+  cfg = config.services.sleepwatcher;
+in
+
+{
+  options = {
+    services.sleepwatcher = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether to enable the sleepwatcher daemon.";
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.darwin.sleepwatcher;
+        description = "This option specifies the sleepwatcher package to use.";
+      };
+
+      on_sleep = mkOption {
+        type = types.nullOr types.lines;
+        default = null;
+        example = "pmset sleepnow";
+        description = "Commands to execute when system goes to sleep.";
+      };
+
+      on_wakeup = mkOption {
+        type = types.nullOr types.lines;
+        default = null;
+        example = "say 'Good morning!'";
+        description = "Commands to execute when system wakes up.";
+      };
+
+      on_display_sleep = mkOption {
+        type = types.nullOr types.lines;
+        default = null;
+        example = "echo 'Display sleeping'";
+        description = "Commands to execute when display goes to sleep.";
+      };
+
+      on_display_wakeup = mkOption {
+        type = types.nullOr types.lines;
+        default = null;
+        example = "echo 'Display awake'";
+        description = "Commands to execute when display wakes up.";
+      };
+
+      on_display_dim = mkOption {
+        type = types.nullOr types.lines;
+        default = null;
+        example = "echo 'Display dimmed'";
+        description = "Commands to execute when display dims.";
+      };
+
+      on_display_undim = mkOption {
+        type = types.nullOr types.lines;
+        default = null;
+        example = "echo 'Display undimmed'";
+        description = "Commands to execute when display undims.";
+      };
+
+      on_idle = mkOption {
+        type = types.nullOr types.lines;
+        default = null;
+        example = "echo 'User idle'";
+        description = "Commands to execute when user becomes idle (requires idleTime).";
+      };
+
+      on_idle_resume = mkOption {
+        type = types.nullOr types.lines;
+        default = null;
+        example = "echo 'User active again'";
+        description = "Commands to execute when user resumes activity after idle.";
+      };
+
+      on_plug = mkOption {
+        type = types.nullOr types.lines;
+        default = null;
+        example = "echo 'Power connected'";
+        description = "Commands to execute when power adapter is plugged in.";
+      };
+
+      on_unplug = mkOption {
+        type = types.nullOr types.lines;
+        default = null;
+        example = "echo 'Power disconnected'";
+        description = "Commands to execute when power adapter is unplugged.";
+      };
+
+      idleTime = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        example = 600;
+        description = "Idle time in seconds before executing on_idle command.";
+      };
+
+      daemonMode = mkOption {
```

Comment:
The naming of this option is misleading, because even when `false` it's still actually a daemon running for the user.
Maybe rename this as `runAsSystem` or `runAsUser` :thinking: